src/config.ts

```diff
@@ -85,6 +85,16 @@ export interface ServiceScriptConfig
    * Whether this service is being invoked directly (e.g. `npm run serve`).
    */
   isDirectlyInvoked: boolean;
+
+  /**
+   * Which scripts depend on this service.
+   *
+   * "Effective" meaning these are not necessarily our direct dependents, since
+   * we include transitive service dependencies through no-command scripts.
```

Comment:
Can you comment on which order these are in since you use "reversed"? I presume this is in deepest dependencies first? And do you need to store it reversed? Can you reverse the array at the use site?

Reply:
The order in the array doesn't matter. This is reverse in the sense of the direction of the edges in the build graph; it's the opposite direction of `effectiveServiceDependencies`.

- `effectiveServiceDependencies`: The services I depend on
- `reverseEffectiveServiceDependencies`: The services that depend on me

I thought about using `dependency` vs `dependent` to distinguish, but I find that it's really easy to misread those.

Any other suggestions for the names?

Reply:
ohhh! Sorry. Yeah, reverse is a little confusing to me. `dependent` might be better even if it's visually close because it's semantically clearer. I would probably default to that unless something clearly better is found. Maybe something with "back" like a back reference would work?

Reply:
Renamed to `services` and `serviceConsumers`